pobieracz_danych_gugik.py

```diff
@@ -1325,37 +1325,56 @@ def model3d_selected_powiat_btn_clicked(self):
         elif self.dockwidget.model3d_lod1_rdbtn.isChecked() and self.dockwidget.model3d_lod2_rdbtn.isChecked():
             standard = ["LOD1", "LOD2"]
         elif not self.dockwidget.model3d_lod1_rdbtn.isChecked() and not self.dockwidget.model3d_lod2_rdbtn.isChecked():
-            msgbox = QMessageBox(QMessageBox.Information, "Ostrzeżenie:",
-                                 f"Nie wybrano standardu")
+            msgbox = QMessageBox(QMessageBox.Information, "Ostrzeżenie:", "Nie wybrano standardu")
             msgbox.exec_()
             return False
 
         od_data_text = self.dockwidget.model3d_dateEdit_comboBox_1.currentText()
         do_data_text = self.dockwidget.model3d_dateEdit_comboBox_2.currentText()
 
-        try:
-            od_data = int(od_data_text)
-        except ValueError:
-            msgbox = QMessageBox(QMessageBox.Warning, "Błąd", "Nieprawidłowa data początkowa.")
-            msgbox.exec_()
-            return False   
-        try:
-            do_data = int(do_data_text)
-        except ValueError:
-            msgbox = QMessageBox(QMessageBox.Warning, "Błąd", "Nieprawidłowa data początkowa.")
-            msgbox.exec_()
-            return False                  
+        def model3d_poprawnosc_dat(od_data_text, do_data_text):
+            current_year = datetime.now().year
+            min_year = 1970
+
+            try:
+                od_data = int(od_data_text)
+                if od_data < min_year or od_data > current_year:
+                    msgbox = QMessageBox(QMessageBox.Warning, "Błąd", f"Data początkowa musi być w przedziale od {min_year} do {current_year}.")
+                    msgbox.exec_()
+                    return False, None, None
+            except ValueError:
+                msgbox = QMessageBox(QMessageBox.Warning, "Błąd", "Nieprawidłowy format daty początkowej.")
+                msgbox.exec_()
+                return False, None, None
+
+            try:
+                do_data = int(do_data_text)
+                if do_data < min_year or do_data > current_year:
+                    msgbox = QMessageBox(QMessageBox.Warning, "Błąd", f"Data końcowa musi być w przedziale od {min_year} do {current_year}.")
+                    msgbox.exec_()
+                    return False, None, None
+            except ValueError:
+                msgbox = QMessageBox(QMessageBox.Warning, "Błąd", "Nieprawidłowy format daty końcowej.")
+                msgbox.exec_()
+                return False, None, None
+
+            return True, od_data, do_data
```

Comment:
Właściwie czemu wykonujesz 2 razy sprawdzenie tej funkcji? Nie wystarczy przed tą funkcją ściągnąć parametr od_data i do_data i wykonać raz takie sprawdzenie? BO trochę niepotrzebnie chyba jest robione to samo 2 razy